components/tidb_query/src/codec/chunk/chunk.rs

```diff
@@ -59,6 +60,77 @@ impl Chunk {
         self.columns[col_idx].append_datum(v)
     }
 
+    /// Append lazy column to the column
+    #[inline]
+    pub fn append_logical_columns(
```

Comment:
Please add some unit tests to cover this exported function.

Reply:
@iosmanthus 
unit tests added